File: python/pdf.py
from playwright.sync_api import sync_playwright

def html_to_pdf(html: str, pdf_path: str):
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()
        page.set_content(html)
        page.pdf(path=pdf_path)
        browser.close()

File: typescript/src/pdf.ts
import playwright from 'playwright';

export async function htmlToPdf(html: string, pdfPath: string) {
    console.log('Launching chrome')
    const browser = await playwright.chromium.launch();
    console.log('hey')
    const page = await browser.newPage();
    await page.setContent(html);
    await page.pdf({ path: pdfPath, format: 'A4' });
    await browser.close();
}